vendor/ember-cache-primitive-polyfill.js

```diff
@@ -0,0 +1,107 @@
+import Ember from 'ember';
+import { DEBUG } from '@glimmer/env';
+import { assert } from '@ember/debug';
+import { gte } from 'ember-compatibility-helpers';
+
+(function() {
+  let track, valueForTag, validateTag, consumeTag, isConstTag;
+
+  if (gte('3.17.0-beta.1')) {
+    let validator = Ember.__loader.require('@glimmer/validator');
+
+    track = validator.track;
+    valueForTag = validator.valueForTag || validator.value;
+    validateTag = validator.validateTag || validator.validate;
+    consumeTag = validator.consumeTag || validator.consume;
+    isConstTag = validator.isConstTag;
+  } else if (gte('3.13.0-beta.1')) {
+    let metal = Ember.__loader.require('@ember/-internals/metal');
+    let reference = Ember.__loader.require('@glimmer/reference');
+
+    track = metal.track;
+    valueForTag = reference.value;
+    validateTag = reference.validate;
+    consumeTag = reference.consume;
+    isConstTag = reference.isConstTag;
+  } else if (DEBUG) {
+    throw new Error('Attempted to use cache polyfill with unsupported Ember version');
+  }
+
+  let DEBUG_CACHE;
+
+  if (DEBUG) {
+    DEBUG_CACHE = Symbol('DEBUG_CACHE');
+  }
+
+  class Cache {
+    lastValue;
+    tag;
+    snapshot = -1;
+
+    constructor(fn) {
+      this.fn = fn;
+
+      if (DEBUG) {
+        return { [DEBUG_CACHE]: this };
+      }
+    }
+  }
```

Comment:
This will need to be tweaked as we iterate on https://github.com/glimmerjs/glimmer-vm/pull/1090